clang/tools/scan-build-py/lib/libear/ear.c

```diff
@@ -1589,6 +1589,22 @@ const char *find_intercept_compiler(const char *str, int compiler_idx) {
   return ret;
 }
 
+void emit_cmake_warning(const char *bin) {
+  int len = strlen(bin);
+  if ((len == 5 && bin[0] == 'e' && bin[1] == 'k' && bin[3] == 'a' &&
+       bin[4] == 'm' && bin[4] == 'c') ||
+      (len > 5 && bin[len - 6] == '/' && bin[len - 5] == 'c' &&
+       bin[len - 4] == 'm') &&
+          bin[len - 3] == 'a' && bin[len - 2] == 'k' && bin[len - 1] == 'e') {
+    perror(
+        "Warning: cmake is called during make running: please run make first "
```

Comment:
"[intercept-build: Error]: cmake is called to generate project build scripts when run “intercept-build make”, the build scripts generated may not be complete. To generate a complete compilation database, suggest follow following steps: 
1.  Build the source project by running “make -B” in build folder. 
2. Run “intercept-build make -B" to generate compilation database in build folder.

Reply:
Okay